libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/NavigationRouteAlternativesRequestCallback.kt

```diff
@@ -19,20 +19,9 @@ interface NavigationRouteAlternativesRequestCallback {
      * The [alternatives] list always represent all available, up-to-date, alternatives for the current route.
      *
      * The alternatives are not automatically added to [MapboxNavigation],
-     * you need to add them manually to trigger [RoutesObserver], for example:
-     * ```kotlin
-     * mapboxNavigation.requestAlternativeRoutes(
-     *     NavigationRouteAlternativesRequestListener { routeProgress, alternatives, routerOrigin ->
-     *         val newRoutes = mutableListOf<NavigationRoute>().apply {
-     *             add(mapboxNavigation.getNavigationRoutes().first())
-     *             addAll(alternatives)
-     *         }
-     *         mapboxNavigation.setRoutes(newRoutes)
-     *     }
-     * )
-     * ```
-     *
-     * You can filter the alternatives out before setting them back to [MapboxNavigation] based on requirements.
+     * you need to add them manually to trigger [RoutesObserver].
+     * The result will also be delivered to [NavigationRouteAlternativesObserver], we recommend you
+     * to filter and set routes there.
```

Comment:
I updated logic so that the callback is called in case of optimisation, but it always has an empty list: https://github.com/mapbox/mapbox-navigation-android/pull/7564/files#diff-485e7c2048009511ed2cb55f1631f02e1cf718c0a97b7173a8b32e457cd1b471R144

I also mentioned in the options documentation: https://github.com/mapbox/mapbox-navigation-android/pull/7564/files#diff-0dd80ae3dbd64ccade9305d4db5a632c3e0bee79191aab735cbf56f835dcff60R26